src/Tickets/Commerce/Order.php

```diff
@@ -514,6 +536,95 @@ public function create( Gateway_Interface $gateway, $args ) {
 		return tec_tc_orders()->set_args( $args )->create();
 	}
 
+	/**
+	 * Filters the values and creates a new Order with Tickets Commerce or updates an existing one.
+	 *
+	 * @since TBD
+	 *
+	 * @param Gateway_Interface $gateway           The gateway to use to create the order.
+	 * @param array             $args              The arguments to create the order.
+	 *
+	 * @return false|WP_Post WP_Post instance on success or false on failure.
+	 */
+	public function upsert( Gateway_Interface $gateway, array $args ) {
+		$gateway_key = $gateway::get_key();
+
+		$existing_order_id = (int) $args['id'] ?? 0;
+		unset( $args['id'] );
+
+		/**
+		 * Allows filtering of the order upsert arguments for all orders created via Tickets Commerce.
+		 *
+		 * @since TBD
+		 *
+		 * @param array             $args    The arguments to create the order.
+		 * @param Gateway_Interface $gateway The gateway to use to create the order.
+		 */
+		$args = apply_filters( "tec_tickets_commerce_order_{$gateway_key}_upsert_args", $args, $gateway );
+
+		/**
+		 * Allows filtering of the order upsert arguments for all orders created via Tickets Commerce.
+		 *
+		 * @since TBD
+		 *
+		 * @param array             $args    The arguments to create the order.
+		 * @param Gateway_Interface $gateway The gateway to use to create the order.
+		 */
+		$args = apply_filters( 'tec_tickets_commerce_order_upsert_args', $args, $gateway );
+
+		/**
+		 * Allows filtering of the existing order ID before "upserting" an order.
+		 *
+		 * @since TDB
+		 *
+		 * @param int $existing_order_id The existing order ID.
+		 */
+		$existing_order_id = (int) apply_filters( 'tec_tickets_commerce_order_upsert_existing_order_id', $existing_order_id );
+
+		if ( ! $existing_order_id || 0 > $existing_order_id ) {
```

Comment:
```suggestion
		if ( ! $existing_order_id || 0 >= $existing_order_id ) {
```
Silly bit of extra security 😉